Remove __init__.py from rootMarkers

mfussenegger opened the pull request:
In python a common folder layout is something like:

    /
    /pkg_name
    /pkg_name/__init__.py
    /pkg_name/main.py
    /setup.py
    /setup.cfg

If `__init__py` is used as root marker the root will be `/pkg_name`.
This is for example problematic for
https://github.com/palantir/python-language-server because it picks up
certain settings from `setup.cfg` or `tox.ini` (among others)


(Maybe it would make sense to add `requirements.txt` or `Pipfile` instead)

mfussenegger committed:
Remove __init__.py from rootMarkers

In python a common folder layout is something like:

    /
    /pkg_name
    /pkg_name/__init__.py
    /pkg_name/main.py
    /setup.py
    /setup.cfg

If `__init__py` is used as root marker the root will be `/pkg_name`.
This is for example problematic for
https://github.com/palantir/python-language-server because it picks up
certain settings from `setup.cfg` or `tox.ini` (among others)

autozimu commented:
> make sense to add requirements.txt or Pipfile instead

Could you add this? Thanks.

mfussenegger committed:
Add Pipfile and requirements.txt python rootMarkers

In case a project lacks a `setup.py` it is still common to have either a
`requirements.txt` or the newer `Pipfile`

autozimu commented:
Thanks.
